tests/slurm_command_gen_strategy/test_common.py

```diff
@@ -28,24 +30,38 @@ def strategy_fixture(slurm_system: SlurmSystem) -> SlurmCommandGenStrategy:
     return strategy
 
 
-def test_filename_generation(strategy_fixture: SlurmCommandGenStrategy, tmp_path: Path):
-    args = {"job_name": "test_job", "num_nodes": 2, "partition": "test_partition", "node_list_str": "node1,node2"}
-    env_vars = {"TEST_VAR": "VALUE"}
-    srun_command = "srun --test test_arg"
-    output_path = tmp_path
+@pytest.fixture
+def testrun_fixture(tmp_path: Path) -> TestRun:
+    mock_test_definition = MagicMock(spec=TestDefinition)
```

Comment:
Why `MagicMock` and not regular `Mock`? Are we using magic methods somehow?

Reply:
Let me see if I can use regular Mock.

Reply:
Updated.